packages/generator/src/edmx-to-vdm/v4/operation.ts

```diff
@@ -120,17 +120,31 @@ export function filterAndTransformOperations(
   operations: EdmxOperation[],
   isBound: boolean
 ): EdmxJoinedOperation[] {
-  const filteredByBoundOperations = operations.filter(
-    operation => (operation.IsBound.toLowerCase() === 'true') === isBound
+  const [withoutOperations, withOperations] = splitOperationsByMissingImports(
+    operationImports,
+    operations
   );
 
-  const [withoutOperation, withOperation] = splitMissingOperation(
-    operationImports,
-    filteredByBoundOperations
+  const relevantOperations = withOperations.filter(
+    operation => operation.IsBound === isBound
   );
 
+  const validBoundOperations =
+    validateBoundOperationParameters(relevantOperations);
+
+  validateOperationImports(
+    withoutOperations,
+    withoutOperations[0]?.operationType
+  );
+
+  return validBoundOperations;
+}
+
+function validateBoundOperationParameters(
+  relevantOperations: EdmxJoinedOperation[]
+): EdmxJoinedOperation[] {
   const [validOperations, operationsWithoutRequiredParameters] =
-    splitMissingParameter(withOperation);
+    splitMissingParameter(relevantOperations);
```

Comment:
nitpick 1: The function is called `splitMissingParameter()` which sounds like a parameter is split. I know you did not introduce it, but it would be great if we could rename it so something speaking, like `splitBoundOperationsByMissingParameters()` or maybe you can think of something shorter.
nitpick 2: The `splitMissingParameter()` function basically ignores unbound operations. I think it should not be called for unbound operations, then the implementation would get simpler.